//  File: log_net.proto
//  Author: Kevin Walsh <kwalsh@holycross.edu>
//
//  Description: Protocol buffers for LogNet.
//
//  Copyright (c) 2014, Kevin Walsh.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package tao;

message LogTimestamp {
	required int32 tm_sec = 1;         /* seconds */
	required int32 tm_min = 2;         /* minutes */
	required int32 tm_hour = 3;        /* hours */
	required int32 tm_mday = 4;        /* day of the month */
	required int32 tm_mon = 5;         /* month */
	required int32 tm_year = 6;        /* year */
	required int32 tm_wday = 7;        /* day of the week */
	required int32 tm_yday = 8;        /* day in the year */
	required int32 tm_isdst = 9;       /* daylight saving time */
}

message LogMessage {
	required int32 severity = 1;
	required string full_filename = 2;
	required string base_filename = 3;
	required int32 line = 4;
	required LogTimestamp time = 5;
	required string message = 6;
}
